yapapi/ctx.py

```diff
@@ -326,6 +96,7 @@ def __init__(
         self._started: bool = False
 
         self.__payment_model: Optional[ComLinear] = None
+        self.__script: Script = self.new_script()
```

Comment:
I would love have all this backward-compatibility-script-thing encapsulated in a single place.

Something like
```

class _BackwardCompatCtxInterface:
    script: Optional[Script]
    
    @deprecated(version="0.7.0", reason="please use a Script object via WorkContext.new_script")
    @classmethod
    def with_script(cls, func):
        @wraps(func)
        def wrapped(*args, **kwargs):
            if cls.script is None:
                cls.script = Script()
            return func(cls.script, *args, **kwargs)

class WorkContext:
    (...)

    @_BackwardCompatCtxInterface.with_script
    def send_file(script, self, src_path, dest_path):
        script.send_file(src_path, dest_path)
```

But I'm not sure if this could be done in a pretty way (what with `commit`?), and even if it could, I'm not sure if this is a good idea.

Reply:
OK.
My only concern is that we might be left with some forever-lasting fallout if we don't clean this well in the next release. But I agree this might not be worth the effort necessary.